features/step_definitions/account_creation_steps.rb

```diff
@@ -0,0 +1,70 @@
+Given(/^I do not have an account on the site$/) do
+end
+
+When(/^I visit the site root path$/) do
+  visit root_path
+end
+
+When(/^I visit the "(.*?)" page$/) do |page|
+  visit page
+end
+
+Then(/^I am presented with a login page$/) do
+  expect(page).to have_content("Log In")
+end
+
+When(/^I click the "(.*?)" link$/) do |link|
+  click_on link
+end
+
+When(/^I enter my email address$/) do
+  fill_in 'Email', with: 'newuser@test.com'
+end
+
+When(/^I enter a password with correct confirmation$/) do
+  fill_in 'Password', with: 'password'
+  fill_in 'Password confirmation', with: 'password'
+end
+
+When(/^I click the "(.*?)" button$/) do |button|
+  click_on button
+end
+
+Then(/^I am told to check my email for a confirmation link$/) do
+  expect(page).to have_content("A message with a confirmation link has been sent to your email address. Please follow the link to activate your account.")
+end
+
+Then(/^I am sent a confirmation email$/) do
+  ActionMailer::Base.deliveries.last.body.match("You can confirm your account email through the link below:")
+end
+
+When(/^I visit the link in that email$/) do
+  ctoken = ActionMailer::Base.deliveries.last.body.match(/confirmation_token=\w*/)
```

Comment:
check out the `email-spec` gem.  i found it very helpful and easy to use.  (you don't have to change what you've done)

https://github.com/email-spec/email-spec